lib/jsdom/living/nodes/Node-impl.js

```diff
@@ -181,19 +181,19 @@ class NodeImpl extends EventTargetImpl {
       newChildImpl._ownerDocument = this._ownerDocument;
     }
 
-    // TODO - if (!newChild) then?
-    if (this.nodeName !== "#document" && newChildImpl._ownerDocument !== this._ownerDocument) {
-      throw new DOMException(DOMException.WRONG_DOCUMENT_ERR);
-    }
-
     if (newChildImpl.nodeType && newChildImpl.nodeType === NODE_TYPE.ATTRIBUTE_NODE) {
       throw new DOMException(DOMException.HIERARCHY_REQUEST_ERR);
     }
 
-    // search for parents matching the newChild
-    for (const ancestor of domSymbolTree.ancestorsIterator(this)) {
-      if (ancestor === newChildImpl) {
-        throw new DOMException(DOMException.HIERARCHY_REQUEST_ERR);
+    if (this._ownerDocument !== newChildImpl._ownerDocument) {
+      // adopt the node when it's not in this document
+      this._ownerDocument.adoptNode(newChildImpl);
+    } else {
+      // search for parents matching the newChild
+      for (const ancestor of domSymbolTree.ancestorsIterator(this)) {
```

Comment:
> Hmm I still don't quite understand why that would be important. Adopting steps only exist for template elements, which I don't see any of in your example.

The also appear in https://dom.spec.whatwg.org/#mutation-algorithms as part of the pre-insertion validity checks?